modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -394,6 +396,52 @@ public PartitionReplicaListener(
         prepareIndexBuilderTxRwOperationTracker();
     }
 
+    @Override
+    public CompletableFuture<Boolean> onPrimaryElected(PrimaryReplicaEventParameters evt) {
+        assert replicationGroupId.equals(evt.groupId()) : format(
+                "The replication group listener does not match the event [grp={}, eventGrp={}]",
+                replicationGroupId,
+                evt.groupId()
+        );
+
+        if (!localNode.id().equals(evt.leaseholderId())) {
+            return falseCompletedFuture();
+        }
+
+        schedulePersistentStorageScan();
+
+        // The future returned by this event handler can't wait for all cleanups because it's not necessary and it can block
+        // meta storage notification thread for a while, preventing it from delivering further updates (including leases) and therefore
+        // causing deadlock on primary replica waiting.
+        return falseCompletedFuture();
+    }
+
+    private void schedulePersistentStorageScan() {
+        txManager.executeWriteIntentSwitchAsync(this::runPersistentStorageScan);
```

Comment:
Why we need to executeWriteIntentSwitchAync()?

Reply:
So, you are using `executeWriteIntentSwitchAync` only in order to have a thread switching side-effect, aren't you?
If true, there should be a better option.